stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/Extensions.kt

```diff
@@ -40,7 +40,13 @@ internal fun Message.users(): List<User> {
  * @param currentUserId The id of the user that the unread count should be evaluated.
  * @param lastMessageAtDate The Date of the last message the SDK is aware of. This is normally the ChannelUserRead.lastMessageSeenDate.
  */
-internal fun Message.shouldIncrementUnreadCount(currentUserId: String, lastMessageAtDate: Date?): Boolean {
+internal fun Message.shouldIncrementUnreadCount(
+    currentUserId: String,
+    lastMessageAtDate: Date?,
+    isChannelMuted: Boolean,
+): Boolean {
+    if (isChannelMuted) return false
```

Comment:
I'm aware checking if the channel is muted can be done sooner at the function call site but I felt like making this a parameter was a good way to ensure we don't forget to factor in if the channel is muted in the future or not.